pkg/patterns/declarative/pkg/manifest/objects_test.go

```diff
@@ -151,3 +154,831 @@ configMapGenerator:
 		})
 	}
 }
+
+func Test_AddLabels(t *testing.T) {
+	inputManifest := `---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: test-app
+  name: frontend
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: test-app
+  strategy: {}
+  template:
+    metadata:
+      labels:
+        app: test-app
+    spec:
+      containers:
+      - image: busybox
+        name: busybox`
+	tests := []struct {
+		name           string
+		inputManifest  string
+		inputLabels    map[string]string
+		expectedLabels map[string]string
+	}{
+		{
+			name:           "add labels which are all new one",
+			inputManifest:  inputManifest,
+			inputLabels:    map[string]string{"sample-key1": "sample-value1", "sample-key2": "sample-value2"},
+			expectedLabels: map[string]string{"app": "test-app", "sample-key1": "sample-value1", "sample-key2": "sample-value2"},
+		},
+		{
+			// If call AddLabels with a key which has exists already, value will be overwritten.
+			name:           "add label which has already exist in manifest",
+			inputManifest:  inputManifest,
+			inputLabels:    map[string]string{"app": "test-app2"},
+			expectedLabels: map[string]string{"app": "test-app2"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			objects, err := ParseObjects(ctx, tt.inputManifest)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			for _, o := range objects.Items {
+				o.AddLabels(tt.inputLabels)
+				if len(tt.expectedLabels) != len(o.object.GetLabels()) {
+					t.Errorf("Expected length of labels to be %v but is %v", len(tt.expectedLabels), len(o.object.GetLabels()))
+				}
+				for k, v := range tt.expectedLabels {
+					if o.object.GetLabels()[k] != v {
+						t.Fatalf("unexpected result, expected ========\n%v\n\nactual ========\n%v\n", tt.expectedLabels, o.object.GetLabels())
```

Comment:
cmp.Diff produces really nice output in this case (`"github.com/google/go-cmp/cmp"`)